src/main/java/datawave/microservice/query/mapreduce/jobs/MapReduceJob.java

```diff
@@ -0,0 +1,202 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.MultiValueMap;
+
+import datawave.core.common.connection.AccumuloConnectionFactory;
+import datawave.core.query.logic.QueryLogicFactory;
+import datawave.microservice.authorization.user.DatawaveUserDetails;
+import datawave.microservice.query.mapreduce.config.MapReduceJobProperties;
+import datawave.microservice.query.mapreduce.config.MapReduceQueryProperties;
+import datawave.microservice.query.mapreduce.status.MapReduceQueryStatus;
+import datawave.webservice.common.audit.AuditParameters;
+import datawave.webservice.query.exception.BadRequestQueryException;
+import datawave.webservice.query.exception.DatawaveErrorCode;
+import datawave.webservice.query.exception.QueryException;
+
+public abstract class MapReduceJob {
+    
+    protected final Logger log = LoggerFactory.getLogger(getClass());
+    
+    protected final MapReduceQueryProperties mapReduceQueryProperties;
+    
+    protected final MapReduceJobProperties mapReduceJobProperties;
+    
+    public MapReduceJob(MapReduceQueryProperties mapReduceQueryProperties) {
+        this.mapReduceQueryProperties = mapReduceQueryProperties;
+        this.mapReduceJobProperties = mapReduceQueryProperties.getJobs().get(this.getClass().getSimpleName());
+    }
+    
+    public String createId(DatawaveUserDetails currentUser) {
+        return UUID.randomUUID().toString();
+    }
+    
+    public void validateParameters(MultiValueMap<String,String> parameters) throws BadRequestQueryException {
+        // Validate the required runtime parameters exist
+        if (null != mapReduceJobProperties.getRequiredRuntimeParameters() && !mapReduceJobProperties.getRequiredRuntimeParameters().isEmpty()
+                        && null != parameters && !parameters.isEmpty()) {
+            // Loop over the required runtime parameter names and make sure an entry exists in the method parameter
+            for (String parameter : mapReduceJobProperties.getRequiredRuntimeParameters().keySet()) {
+                if (!parameters.containsKey(parameter) || StringUtils.isBlank(parameters.getFirst(parameter))) {
+                    log.error("Required runtime parameter '" + parameter + "' must be set");
+                    throw new BadRequestQueryException("Required runtime parameter '" + parameter + "' must be set.", HttpStatus.SC_BAD_REQUEST + "-1");
+                }
+            }
+        }
+        
+        parameters.remove(AuditParameters.QUERY_SECURITY_MARKING_COLVIZ);
+        parameters.remove(AuditParameters.USER_DN);
+        parameters.remove(AuditParameters.QUERY_AUDIT_TYPE);
+    }
+    
+    /**
+     * @return HDFS FileSystem
+     * @throws IOException
+     */
+    protected final FileSystem getFileSystem(Configuration conf) throws IOException {
+        return FileSystem.get(conf);
+    }
+    
+    /**
+     * Common code to setup distributed cache and classpath for the job
+     *
+     * @param job
```

Comment:
add @param baseDir, descriptions for the params & throws

Reply:
sure.